services/api-server/tests/unit/test_api__study_workflows.py

```diff
@@ -0,0 +1,318 @@
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+        return JobStatus(**resp.json())
+
+    @_handle_http_status_error
+    async def get_study_job_outputs(self, study_id, job_id) -> JobOutputs:
+        resp = await self._client.post(
+            f"{API_VTAG}/studies/{study_id}/jobs/{job_id}/outputs",
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+        return JobOutputs(**resp.json())
+
+    @_handle_http_status_error
+    async def delete_study_job(self, study_id, job_id) -> None:
+        resp = await self._client.delete(
+            f"{API_VTAG}/studies/{study_id}/jobs/{job_id}",
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+
+
+@pytest.fixture
+def input_json_path(tmp_path: Path) -> Path:
+    # https://github.com/wvangeit/osparc-pyapi-tests/blob/master/noninter1/input.json
+    p = tmp_path / "input.json"
+    data = {"f1": 3}
+    p.write_text(json.dumps(data))
+    return p
+
+
+@pytest.fixture
+def input_data_path(tmp_path: Path) -> Path:
+    p = tmp_path / "input.data"
+    data = {"x1": 5.0, "y2": 7.5}
+    p.write_text(json.dumps(data))
+    return p
+
+
+@pytest.fixture
+def test_py_path(tmp_path: Path) -> Path:
+    p = tmp_path / "test.py"
+    code = textwrap.dedent(
+        """\
+    import os
+    import json
+
+    from pathlib import Path
+
+    print("Konichiwa. O genki desu ka?")
+
+    input_path = Path(os.environ["INPUT_FOLDER"])
+    output_path = Path(os.environ["OUTPUT_FOLDER"])
+
+    test_data_path = input_path / "input.data"
+
+    test_data = json.loads(test_data_path.read_text())
+
+    output_paths = {}
+    for output_i in range(1, 6):
+        output_paths[output_i] = output_path / f"output_{output_i}"
+
+        output_paths[output_i].mkdir(parents=True, exist_ok=True)
+
+    output_data_path = output_paths[1] / "output.data"
+
+    output_data_path.write_text(json.dumps(test_data))
+
+    print(f"Wrote output files to: {output_path.resolve()}")
+
+    print("Genki desu")
+    """
+    )
+    p.write_text(code)
+    return p
+
+
+class MockedBackendApiDict(TypedDict):
+    webserver: MockRouter | None
+    storage: MockRouter | None
+    director_v2: MockRouter | None
+
+
+@pytest.fixture
+def mocked_backend(
+    project_tests_dir: Path,
+    mocked_webserver_service_api_base: MockRouter,
+    mocked_storage_service_api_base: MockRouter,
+    mocked_directorv2_service_api_base: MockRouter,
+    respx_mock_from_capture: Callable[
+        [list[respx.MockRouter], Path, list[SideEffectCallback]],
+        list[respx.MockRouter],
+    ],
+    mocker: MockerFixture,
+) -> MockedBackendApiDict | None:
+    respx_mock_from_capture(
+        [
+            mocked_webserver_service_api_base,
+            mocked_storage_service_api_base,
+            mocked_directorv2_service_api_base,
+        ],
+        project_tests_dir / "mocks" / "run_study_workflow.json",
+        [],
+    )
```

Comment:
So it works just like this 😓. Nice 😄

Reply:
If you want to be an over-achiever like me you could add some sideeffects which check that the various uuids end up where they are expected to end up. Just a suggestion

Reply:
challenge accepted :-D will check in following PRs